### ISSUE TITLE
Error handling section content is not related to error/exception

### ISSUE THREAD

**Issue by avitsidis** [thangchung/clean-code-dotnet#30]
It seems to me that the error handling section is a bit messy. Why are the *Use consistent capitalization*
and *Function callers and callees should be close* in this section? IMHO, *Function callers and callees should be close* should be in the Functions section and *Use consistent capitalization* should be in the naming section. 

Also, a lot of developer do not understand the difference between `throw e;` and `throw;`. I can write a piece of code about it to fill the Error handling section.

**Comment by Justbeingjustin** [comment]
> Also, a lot of developer do not understand the difference between throw e; and throw;. 

So true! 

I remember the day that I read [this](https://stackoverflow.com/a/2999314/8846030).

**Comment by Justbeingjustin** [comment]
Maybe something like this could be a good start: 

The stack trace is preserved with throw;


Bad
```
try
{
    // code & logic
}
catch (Exception ex)
{
    throw ex;
}
```

Good
```
try
{
    // code & logic
}
catch (Exception ex)
{
    throw;
}
```

**Comment by avitsidis** [comment]
@Justbeingjustin your samples are correct but a second good sample could be whenever you want to throw your own exception:
```
try
{
    // code & logic
}
catch (Exception ex)
{
    throw new CustomException("some information about the exception", e);
}
```
Here is a text we could add to a new sub section _Rethrowing exceptions_ in Error Handling section:

> ### Rethrowing exceptions
> 
> C# allows you to re-throw an exception in a catch block using the `throw` keyword. However, it is (most of the time) a bad practice to throw a caught exception using `throw e;`. This statement will re-throw your exception but it will also reset the stack trace of that exception. If your goal is to re-throw the exact same exception, use `throw;` so that you keep your stack trace unchanged. If you want to enrich with a custom exception, then you can instantiate a new exception and set its inner exception property with the caught exception with `throw new CustomException("some info", e);`. 
> 
> Adding information to an exception is a good practice as it will help when debugging. However if you only want to log or notify an exception happened just use `throw;` to pass the buck to the caller after you log/notify it.
>
> {samples inserted here}

I'm not an English native, can you tell me if this text is correct technically and grammatically speaking!

If it is correct, I'll do a pull request for this :-]

**Comment by Justbeingjustin** [comment]
@avitsidis it appears that there are two issues here.

1) Resort a few of the entries 
2) Add a new throw entry to the error handling section

 Let's move the throw entry conversation to a new issue. #35

**Comment by thangchung** [comment]
I closed it for now and move to #36
